### PR TITLE
GH-38670: [Release] Add missing `r/pkgdown/assets/versions.html` to update versions

### LINKED ISSUE #38670: [Release] `r/pkgdown/assets/versions.html` is missing when updating versions
### Describe the bug, including details regarding any error messages, version, and platform.

As part of the post-release tasks we bump-versions. When I use the bump-versions there is one file that has been modified but is not committed as part of the script.

We should add the missing file:
```
$ git diff
diff --git a/r/pkgdown/assets/versions.html b/r/pkgdown/assets/versions.html
index 8ba513a..5ef480b 100644
--- a/r/pkgdown/assets/versions.html
+++ b/r/pkgdown/assets/versions.html
@@ -1,9 +1,12 @@
 <!DOCTYPE html>
 <html>
-<body><p><a href="../dev/r/">13.0.0.9000 (dev)</a></p>
-<p><a href="../r/">13.0.0.1 (release)</a></p>
+<body><p><a href="../dev/r/">14.0.1.9000 (dev)</a></p>
+<p><a href="../r/">14.0.1 (release)</a></p>
+<p><a href="../14.0/r/">14.0.0</a></p>
+<p><a href="../13.0/r/">13.0.0.1</a></p>
 <p><a href="../12.0/r/">12.0.1.1</a></p>
-<p><a href="../11.0/r/">11.0.0.3</a></p>
+<p><a href="../12.0/r/">12.0.0</a></p>
+<p><a href="../11.0/r/">11.0.0</a></p>
 <p><a href="../10.0/r/">10.0.1</a></p>
 <p><a href="../9.0/r/">9.0.0</a></p>
 <p><a href="../8.0/r/">8.0.0</a></p>
```

### Component(s)

Release

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

File is missing when bumping versions.

### What changes are included in this PR?

Remove the files as it seems is not used.

### Are these changes tested?

Not necessary

### Are there any user-facing changes?

No
* Closes: #38670

### CODE REVIEW

File: r/pkgdown/assets/versions.html [hunk 1]
```diff
@@ -1,9 +1,11 @@
 <!DOCTYPE html>
 <html>
-<body><p><a href="../dev/r/">13.0.0.9000 (dev)</a></p>
-<p><a href="../r/">13.0.0.1 (release)</a></p>
+<body><p><a href="../dev/r/">14.0.0.9000 (dev)</a></p>
+<p><a href="../r/">14.0.0 (release)</a></p>
+<p><a href="../13.0/r/">13.0.0.1</a></p>
 <p><a href="../12.0/r/">12.0.1.1</a></p>
-<p><a href="../11.0/r/">11.0.0.3</a></p>
+<p><a href="../12.0/r/">12.0.0</a></p>
```
Comment (kou) [inline r/pkgdown/assets/versions.html:7]: Should we remove this?

**Review by kou — APPROVED** [review]
+1

File: r/pkgdown/assets/versions.html (hunk 1 above)
Comment (raulcd) [reply r/pkgdown/assets/versions.html:7]: This is part of the automated diff generated when `post-11-bump-versions.sh` is run. I also found that strange but might mean there is something wrong when bumping the versions.
@thisisnic @paleolimbot @assignUser is there anything we should fix when automatically bumping versions on the `versions.html` file?

**Comment by raulcd** [comment]
@assignUser @paleolimbot as discussed I've removed the file as it seems not used.

**Comment by raulcd** [comment]
I reverted the last change because I thought the R failures could be related but they don't seem to be as they are failing without removing the file.

File: r/pkgdown/assets/versions.html (hunk 1 above)
Comment (assignUser) [reply r/pkgdown/assets/versions.html]: I'm not familiar with the doc setup so :shrug: sorry^^

**Comment by thisisnic** [comment]
> @assignUser @paleolimbot as discussed I've removed the file as it seems not used.

The file is not needed in the repo; it's just needed for the website, as long as it has the correct versions here: https://arrow.apache.org/docs/r/versions.html

**Comment by thisisnic** [comment]
I think the comments above mean that this PR is no longer needed so I'm closing this, but if I'm wrong let me know and we can reopen and merge this

**Pull request closed without merging**